src/operators/custom_version.rs

```diff
@@ -0,0 +1,114 @@
+use std::cmp::Ordering;
+use std::collections::VecDeque;
+use unicode_normalization::UnicodeNormalization;
+
+fn split_version(version: &str) -> VecDeque<String> {
+    let mut components = VecDeque::new();
+    let mut part = String::new();
+    for ch in version.chars() {
+        if ch == '.' || ch == '-' {
+            components.push_back(part.drain(..).collect());
+            part = String::new();
+        } else {
+            part.push(ch);
+        }
+    }
+    components.push_back(part.drain(..).collect());
+    components
+}
+
+pub fn compare_version(str1: &str, str2: &str, numeric: bool) -> Ordering {
+    let norm_str1 = str1.nfc().collect::<String>();
+    let norm_str2 = str2.nfc().collect::<String>();
+    let mut iter1 = split_version(&norm_str1);
+    let mut iter2 = split_version(&norm_str2);
+
+    loop {
+        let comp1 = iter1.pop_front();
+        let comp2 = iter2.pop_front();
+
+        match (comp1, comp2) {
+            (None, None) => return Ordering::Equal,
+            (Some(ref _s1), None) => return Ordering::Greater,
+            (None, Some(ref _s2)) => return Ordering::Less,
+            (Some(s1), Some(s2)) => {
+                let order = if numeric {
+                    match (s1.parse::<u32>(), s2.parse::<u32>()) {
+                        (Ok(i1), Ok(i2)) => i1.cmp(&i2),
+                        _ => s1.cmp(&s2),
```

Comment:
If you can't parse into an integer shouldn't you throw an error?

Reply:
the version supports alpha numeric characters, so its not an error actually, I will rename the parameter from numeric to alpha_numeric to avoid this confusion